api/ocm/plugin/internal/transfer.go

```diff
@@ -0,0 +1,128 @@
+	Version    string                    `json:"version"`
+	Provider   metav1.Provider           `json:"provider,omitempty"`
+	Repository ocm.GenericRepositorySpec `json:"repository"`
+	Labels     metav1.Labels             `json:"labels,omitempty"`
+}
+
+type TargetRepositorySpec = ocm.GenericRepositorySpec
+
+// TransferOptions are the standard transfer options from
+// the standard transfer handler.
+// Like other transfer handler types it is possible to define
+// more options. Those non-standard options are passed
+// via a json.RawMessage in the field special.
+type TransferOptions struct {
+	Recursive         *bool            `json:"recursive,omitempty"`
+	ResourcesByValue  *bool            `json:"resourcesByValue,omitempty"`
+	LocalByValue      *bool            `json:"localByValue,omitempty"`
+	SourcesByValue    *bool            `json:"sourcesByValue,omitempty"`
+	KeepGlobalAccess  *bool            `json:"keepGlobalAccess,omitempty"`
+	StopOnExisting    *bool            `json:"stopOnExisting,omitempty"`
+	EnforceTransport  *bool            `json:"enforceTransport,omitempty"`
+	Overwrite         *bool            `json:"overwrite,omitempty"`
+	SkipUpdate        *bool            `json:"skipUpdate,omitempty"`
+	OmitAccessTypes   []string         `json:"omitAccessTypes,omitempty"`
+	OmitArtifactTypes []string         `json:"omitArtifactTypes,omitempty"`
+	Special           *json.RawMessage `json:"special,omitempty"`
+}
+
+// Resolution describes the transport context for a component
+// version, including the new handler specification and
+// the source repository to use to look up the reference.
+type Resolution struct {
+	RepositorySpec *ocm.GenericRepositorySpec `json:"repository,omitempty"`
+	// TransferHandler is the handler identity according to the transfer handler
+	// name scheme.
+	TransferHandler string `json:"transferHandler,omitempty"`
+	// TransferOptions may describe modified options used for sub-sequent
+	// transfers.
+	TransferOptions *TransferOptions `json:"transferOptions,omitempty"`
+}
+
+// DecisionRequestResult is the structure of the answer
+// the plugin has to return for a question.
+type DecisionRequestResult struct {
+	Error      string      `json:"error,omitempty"`
+	Decision   bool        `json:"decision"`
+	Resolution *Resolution `json:"resolution,omitempty"`
+}
+
+// Question is the interface for the question attributes
+// differing for the various questions types.
+// There are three basic attribute sets:
+//   - ComponentVersionQuestion
+//   - ComponentReferenceQuestion
+//   - ArtifactQuestion
+//
+// For type assignments see TransferHandlerQuestions.
+type Question interface{}
+
+// ComponentVersionQuestion describes the question arguments
+// given for a component version related question.
+type ComponentVersionQuestion struct {
+	Source  SourceComponentVersion `json:"source"`
+	Target  TargetRepositorySpec   `json:"target"`
+	Options TransferOptions        `json:"options"`
+}
```

Comment:
Significantly better than before. But I'd still like to see a correlation of the `Question` interface to the actual question types (e.g. implement methods for asserting `AsComponentVersionQuestion() (*ComponentVersionQuestion, error)`, `AsArtifactQuestion() (*ArtifactQuestion, error)`). 
WDYT?

Reply:
The types are just value container, there are no methods. I could add a dummy methods to be able to find implementations.

Reply:
Added method QuestionArgumentsType